### ISSUE TITLE
keyboard close animation

### ISSUE THREAD

**Issue by v1z** [Mottie/Keyboard#605]
Hello, is it possible to add an animation when keyboard is closing?
Something like that on appereance - http://jsfiddle.net/egb3a1sk/2966/
But I want keyboard to fly away (not just disappear) on closing.

**Comment by Mottie** [comment]
Hi @v1z!

Currently, the keyboard is immediately destroyed on close, so there is no way to animate "out" the keyboard. Using the [`beforeClose` option](https://github.com/Mottie/Keyboard/wiki/Methods#beforeclose---function) won't work because it is treated as an event (a return value is not expected) and not a callback.

So, I'll need to think about this more but I could add a `close` callback option. It'll require that you remove the keyboard container after the animation completes or you'll get a memory leak... something like this:

```js
close: function(keyboard, $keyboardElm) {
  // keyboard.$keyboard usually contains the $keyboardElm
  // but the reference would need to be removed (set to
  // an empty array `[]`) to prevent problems within the
  // code that checks the keyboard.$keyboard size
  keyboardElm.hide('bounce', function() {
    keyboardElm.remove();
  });
}
```

**Comment by v1z** [comment]
Yeah, it would be nice to have an option allowing animate keyboard out. 
It will allow to build UI like on tablets/mobiles - where the keyboard is sliding in/out.

**Comment by vkurlyan** [comment]
Maybe it would be better to have argument 'destroyKeyboardFn' in the close event, so developers will not depend on internal implementation:

```
close: function(keyboard, $keyboardElm, destroyKeyboardFn) {
  keyboardElm.hide('bounce', function() {
    // Destroy keyboard properly
    destroyKeyboardFn();
  });
}
```
